### PR TITLE
one-one mapping for user and userpii

### PR DESCRIPTION
**Pull request by Jaswitha**

### CODE REVIEW

**Comment by Jaswitha** [comment]
pushing the changes for user and userpii

File: pom.xml [hunk 1]
```diff
@@ -105,7 +105,11 @@
   	<artifactId>spring-security-taglibs</artifactId>
   	<version>4.0.2.RELEASE</version>
   </dependency>
-  
+  <dependency>
+    <groupId>org.springframework</groupId>
+    <artifactId>spring-context-support</artifactId>
+    <version>3.2.8.RELEASE</version>
```
Comment (chasethenag420) [inline pom.xml]: Why version 3? We are using 4 for springs

File: sql/test_user.sql [hunk 2]
```diff
@@ -31,13 +31,17 @@ CREATE TABLE `user` (
   `USERNAME` varchar(30) NOT NULL,
   `PASSWORD` varchar(15) DEFAULT NULL,
   `ROLEID` int(20) DEFAULT NULL,
+
+  `userstatus` varchar(20) DEFAULT NULL,
+
```
Comment (chasethenag420) [inline sql/test_user.sql]: Why two columns with same name an different length

Comment (Jaswitha) [inline sql/test_user.sql]: In the first case you can remove that .I forgot to remove it.

In the second case where did you find 2 columns ? which columns u are
talking about?

Comment (chasethenag420) [inline sql/test_user.sql]: `userstatus` varchar(20) DEFAULT NULL,

  `USERSTATUS` varchar(7) DEFAULT NULL,

File: src/main/java/com/asu/cse545/group12/controller/UserController.java [hunk 3]
```diff
@@ -63,13 +77,30 @@ public ModelAndView getForm() {
...
 		
 		return modelView;
 	}
 
-	@RequestMapping("signUpExternalUser")
+	@RequestMapping( value= "signUpExternalUser")
+
+	public ModelAndView registerUser(@ModelAttribute Users user) {
+		//Users user=(Users)modelMap.get("user");
+		user.setUserStatus("A");
```
Comment (chasethenag420) [inline src/main/java/com/asu/cse545/group12/controller/UserController.java]: Does A indicates active? we might want to insert record for signup as inactive once approved we can set it to active user.

File: sql/test_user.sql [hunk 4]
```diff
@@ -31,13 +31,17 @@ CREATE TABLE `user` (
...
   `USERSTATUS` varchar(7) DEFAULT NULL,
+
   `REGISTRATION_DATE` date DEFAULT NULL,
   `LAST_MODIFIED_DATE` date DEFAULT NULL,
   `EMAILID` varchar(30) NOT NULL,
   `PHONENUM` varchar(15) NOT NULL,
   PRIMARY KEY (`USERID`)
-) ENGINE=InnoDB AUTO_INCREMENT=2 DEFAULT CHARSET=latin1;
+) ENGINE=InnoDB AUTO_INCREMENT=3 DEFAULT CHARSET=latin1;
```
Comment (chasethenag420) [inline sql/test_user.sql]: Charset shouldn't we use UTF-8

File: src/main/java/com/asu/cse545/group12/controller/UserController.java [hunk 5]
```diff
@@ -63,13 +77,30 @@ public ModelAndView getForm() {
...
+		java.util.Date date = Calendar.getInstance().getTime();
+		user.setLastModifieddate(date);
+		user.setRegistrationDate(date);
+		System.out.println("9999"+user.getUserpii());
+		user.setUserId(user.getUserpii().getUserId());
+		user.getUserpii().setDateOfBirth(date);
```
Comment (chasethenag420) [inline src/main/java/com/asu/cse545/group12/controller/UserController.java]: This logic of setting user fields should actually be done in service layer. controller should not deal with this. instead controller will do validation stuff and once valid it should call service layer to do this logic

File: src/main/java/com/asu/cse545/group12/dao/UserDaoImpl.java [hunk 6]
```diff
@@ -30,6 +30,7 @@ public int insertRow(Users user) {
 		Transaction tx = session.beginTransaction();  
 		user.setUserStatus("InActive");
```
Comment (chasethenag420) [inline src/main/java/com/asu/cse545/group12/dao/UserDaoImpl.java]: above we are use 'A' here we are using 'Inactive' lets stick to acitve or inactive. And this logic should not be here move it to service layer. Dao will deal only with DB setings and getting should not update any data.

File: src/main/java/com/asu/cse545/group12/domain/UserPII.java [hunk 7]
```diff
@@ -1,17 +1,39 @@
...
-	private int userId;
+@Entity(name="userpii")
+public class UserPII implements Serializable{
 	
+	
+	
+			@Id
+			@Column(name="USERID")
+			@GeneratedValue(strategy = GenerationType.IDENTITY) 
+			private int userId;
```
Comment (chasethenag420) [inline src/main/java/com/asu/cse545/group12/domain/UserPII.java]: Im not sure if there is way..can u check if can be declared as functional key

File: src/main/java/com/asu/cse545/group12/domain/Users.java [hunk 8]
```diff
@@ -85,7 +102,7 @@ public void setUserId(int userId) {
 		this.userId = userId;
 	}
 
-
+	@Column(name = "FIRSTNAME")
```
Comment (chasethenag420) [inline src/main/java/com/asu/cse545/group12/domain/Users.java]: Is there a reason why each get method is annotated?

**Pull request merged**